### ISSUE TITLE
FILTER vs CQL_FILTER depuis un flux QGIS Server

### ISSUE THREAD

**Issue by vfabry** [geobretagne/mviewer#334]
Bonjour,
QGIS Server n'utilise pas le paramètre `cql_filter` dans le requêtes GetMap, mais le paramètre `filter`.
Par conséquent, on ne peut pas utiliser le paramètre `filter` de mviewer, ni `attributefilter`.
Vous voyez un moyen de contourner ça ?

Vincent
Géo2France

**Comment by spelhate** [comment]
Oui. Il faudrait une petite évolution de mviewer pour via la config définir le nom du paramètre permettant de filtrer côté serveur.
Et ça marche aussi avec du CQL ?

**Comment by vfabry** [comment]
non, malheureusement, ça ne marche pas avec du CQL...
+1 pour l'évolution côté mviewer :-)
@bchartier tu saurais faire ça ?

**Comment by Gaetanbrl** [comment]
@spelhate  @vfabry est-ce que la syntaxe est la même entre le CQL et le Filter Qgis Server ?

Si oui, on peut simplement rajouter un paramètre "server" afin de saisir la valeur qgis ou geoserver (par défaut - si non ajouté dans le config).

Comme cela le paramètre "filter" déjà existant pourra simplement renommer le paramètre dans le requête.

**Comment by vfabry** [comment]
J'ai l'impresison que ce n'est pas tout à fait la même syntaxe (notamment au niveau des doubles quotes pour les champs). Voici ce que dit la doc QGIS Server : 

> FILTER parameter: Subsets of layers can be selected with the FILTER parameter. Syntax is basically the same as for the QGIS subset string. However, there are some restrictions to avoid SQL injections into databases via QGIS server:
> 
> Text strings need to be enclosed with quotes (single quotes for strings, double quotes for attributes) A space between each word / special character is mandatory. Allowed Keywords and special characters are ‘AND’,’OR’,’IN’,’=’,’<’,’>=’, ‘>’,’>=’,’!=*,’(‘,’)’. Semicolons in string expressions are not allowed

**Comment by Gaetanbrl** [comment]
En fait pas besoin de similarité (je ne sais plus pourquoi je me suis posé la question...).

On peut partir du principe que la saisie du paramètre `server` oblige que le `filter` soit de la syntaxe du server saisi.

Si `server = "qgis"` alors la syntaxe est forcément adaptée pour Qgis Server, etsi `server = "geoserver"` ou pas de paramètre `server`, alors c'est du CQL ?

En fait ça me paraît même logique. Derrière, il faudra seulement modifier le nom du paramètre dans le coeur.

Par contre comme ça, je pense qu'il faudrait vérifier que la requête soit similaire dans sa construction ou voir s'il y a des adaptations à faire.

**Comment by spelhate** [comment]
Tout à fait d'accord avec @Gaetanbrl , dans le filtre, on met l'expression qui est compatible avec le serveur. Par contre, il faut disposer du nom du paramètre  à adresser au serveur 
 * CQL_FILTER pour geoserver (valeur par défaut)
 * FILTER pour QGIS server
 * ??? pour MAPSERVER
Il faudrait donc ajouter un nouveau paramètre de configuration pour les couches OGC : **filterparam** ?

**Comment by Gaetanbrl** [comment]
@spelhate `filterparam `serait utile pour ajouter les paramètres de la requête ? (bbox, epsg, version etc. etc.) ?

**Comment by Gaetanbrl** [comment]
Un complément technique de comparaison :

https://docs.qgis.org/3.22/pdf/fr/QGIS-3.22-ServerUserGuide-fr.pdf

![image](https://github.com/user-attachments/assets/ad980e06-f95c-49e1-956d-64f54d830a2b)

**Comment by Gaetanbrl** [comment]
En complément, je viens de voir un paramètre WMS OpenLayers pour spécifier le type du serveur :
https://openlayers.org/en/latest/apidoc/module-ol_source_wms.html#~ServerType

<img width="615" height="188" alt="Image" src="https://github.com/user-attachments/assets/b909395b-45f4-47ee-9a26-03c1eeff5459" />
